uni/lib/model/providers/lazy/restaurant_provider.dart

```diff
@@ -39,7 +44,7 @@ class RestaurantProvider extends StateProviderNotifier {
       updateStatus(RequestStatus.busy);
 
       final List<Restaurant> restaurants =
-          await RestaurantFetcher().getRestaurants(session);
+      await RestaurantFetcher().getRestaurants(session);
       // Updates local database according to information fetched -- Restaurants
       final RestaurantDatabase db = RestaurantDatabase();
       db.saveRestaurants(restaurants);
```

Comment:
Besides loading from the database, you need to load the saved favourites into the provider here. If you reload the app now, it's like the favourites have disappeared, but they are still in the shared preferences.